pages/docs/reference/reflection.md

```diff
@@ -11,13 +11,57 @@ Reflection is a set of language and library features that allows for introspecti
 Kotlin makes functions and properties first-class citizens in the language, and introspecting them (i.e. learning a name or 
 a type of a property or function at runtime) is closely intertwined with simply using a functional or reactive style.
 
-> On the Java platform, the runtime component required for using the reflection features is distributed as a separate
-JAR file (`kotlin-reflect.jar`). This is done to reduce the required size of the runtime library for applications
-that do not use reflection features. If you do use reflection, please make sure that the .jar file is added to the
-classpath of your project.
+> On the JavaScript platform, only class references are currently supported. [Learn more about reflection in Kotlin/JS](js-reflection.html).
 {:.note}
 
-## Class References
+## JVM dependency
+
+On the JVM platform, the runtime component required for using the reflection features is distributed as a separate
+artifact. This is done to reduce the required size of the runtime library for applications
+that do not use reflection features. If you do use reflection, add the dependency on `kotlin-reflect` to your project:
+* In Gradle:
+  <div class="multi-language-sample" data-lang="groovy">
+  <div class="sample" markdown="1" theme="idea" mode="groovy" data-highlight-only>
+  
+  ```groovy
+  dependencies {
+      compile "org.jetbrains.kotlin:kotlin-reflect:{{ site.data.releases.latest.version }}"
+  }
+  ```
+  
+  </div>
+  </div>
+   
+  <div class="multi-language-sample" data-lang="kotlin">
+  <div class="sample" markdown="1" theme="idea" mode="kotlin" data-highlight-only>
+  
+  ```kotlin
+  dependencies {
+    implementation("org.jetbrains.kotlin:kotlin-reflect:{{ site.data.releases.latest.version }}")
+  }
+  ```
+  
+  </div>
+  </div>
+  
+  * In Maven:
+
+  <div class="sample" markdown="1" mode="xml" auto-indent="false" theme="idea" data-highlight-only>
+    
+  ```xml
+  <dependencies>
+      <dependency>
+          <groupId>org.jetbrains.kotlin</groupId>
+          <artifactId>kotlin-reflect</artifactId>
+      </dependency>
+  </dependencies>
+  ```
+  </div>
+
+If you don't use Gradle or Maven, you can add the `kotlin-reflect.jar` file from the Kotlin compiler distribution to
```

Comment:
If one doesn't use Gradle or Maven, they're likely using either JPS or CLI (or Ant), and in all those cases `kotlin-reflect` is added by default (there's a flag `-no-reflect` to prevent this). Maybe rephrase this to something like "ensure you have kotlin-reflect.jar in the classpath"?